pytorch_ops/generate_tests.py

```diff
@@ -0,0 +1,201 @@
+        running the test.
+        """
+        inputs = self.generate_inputs()
+        self.test_config["rtol"] = rtol
+        self.test_config["atol"] = atol
+        self.test_config["equal_nan"] = equal_nan
+        self.save_mlir(*inputs)
+        expected_results = self.generate_expected_value(*inputs)
+        self.save_inputs(*inputs)
+        self.save_results(*expected_results)
+        self.save_config()
+
+    def generate_expected_value(self, *args):
+        results = self.forward(*args)
+        return [results]
+
+    def generate_inputs(self):
+        return self.args
+
+    @abstractmethod
+    def forward(self, *args):
+        ...
+
+
+@dataclass
+class RandomSession:
+    generator: torch.Generator = torch.Generator()
+    seed: int = 0
+
+    def __post_init__(self):
+        self.generator.manual_seed(self.seed)
+
+    def rand(self, *args, **kwargs):
+        if kwargs.get("generator"):
+            raise ValueError("RandomSession uses its own generator")
+        kwargs["generator"] = self.generator
+        return torch.rand(*args, **kwargs)
+
+
+class AB(TestGenerator):
+    def forward(self, left, right):
+        return left @ right
+
+
+class AB_bfloat16(TestGenerator):
+    def forward(self, left, right):
+        left = left.to(torch.bfloat16)
+        right = right.to(torch.bfloat16)
+        res = left @ right
+        return res.to(torch.float32)
+
+
+class ATB(TestGenerator):
+    def forward(self, left, right):
+        return left.t() @ right
+
+
+class ABT(TestGenerator):
+    def forward(self, left, right):
+        return left @ right.t()
+
+
+class ABplusC(TestGenerator):
+    def forward(self, A, B, C):
+        return A @ B + C
+
+
+class ReluABPlusC(TestGenerator):
+    def forward(self, A, B, C):
+        return torch.relu(A @ B + C)
+
+
+class GeluABPlusC(TestGenerator):
+    def forward(self, A, B, C):
+        return torch.ops.aten.gelu.default(A @ B + C)
+
+
+# TODO: Future PR, itertools.product where it makes sense
+for dtype in [torch.float32]:
+    for cls in [AB_bfloat16]:
+        random = RandomSession()
+        inputs = (random.rand(64, 64, dtype=dtype), random.rand(64, 64, dtype=dtype))
+        dyn_dim = torch.export.Dim("N")
+        dynamic_shapes = {"left": {0: dyn_dim}, "right": {1: dyn_dim}}
+        instance = cls(
+            *inputs,
+            name=cls.__name__,
+            export_kwargs={"dynamic_shapes": dynamic_shapes},
+        )
+        instance.generate_test(atol=1e-2, rtol=1e-2)
+for dtype in [torch.float32, torch.float16]:
+    for cls in [AB]:
+        random = RandomSession()
+        inputs = (random.rand(64, 64, dtype=dtype), random.rand(64, 64, dtype=dtype))
+        dyn_dim = torch.export.Dim("N")
+        dynamic_shapes = {"left": {0: dyn_dim}, "right": {1: dyn_dim}}
+        instance = cls(
+            *inputs,
+            name=cls.__name__ + str(dtype),
+            export_kwargs={"dynamic_shapes": dynamic_shapes},
+        )
```

Comment:
Uhhh, this is not how it should be. Dynamic dimensions should not be added at class initialization time. It should be a class property. Instead implement a `def get_dynamic_shapes` in TestGenerator class and ask the user to override it.

The default implementation can be empty, which is everything is static.